add a FG RootCAConfigMap and update timeline (#2128)
kubernetes/enhancements: keps/sig-auth/1205-bound-service-account-tokens/README.md

1.  Add annotation to audit events for legacy and stale tokens including
    necessary information to locate problematic client.

> 

Reviewer: above, note the `--service-account-extend-token-expiration` mitigation defaults to true, and that cluster administrators can set it to `--service-account-extend-token-expiration=false` to turn off the mitigation if desired

Author: done

### Test Plan

#### TokenRequest/TokenRequestProjection

- Unit tests
- E2E tests
  - Projected jwt tokens are correctly mounted. (conformance test)
  - The owner and mode of projected tokens are correctly set
  - In-cluster clients work with Token rotation

#### RootCAConfigMap

> 

Reviewer: Add a description of tests for this feature gate

Author: done

#### BoundServiceAccountTokenVolume

- Unit tests
-  An upgrade test
  1. Create pod A with feature disabled where pod A is working and a secret volume
    is mounted
  2. Enable feature where pod A continue working
  3. Create pod B and it is working and projected volumes are mounted

### Graduation Criteria

#### TokenRequest/TokenRequestProjection

| Alpha | Beta | GA |
| --- | --- | --- |
| 1.10 | 1.12 | 1.20 |

##### Alpha->Beta

> 

Reviewer: drop the Alpha->Beta section

Author: done

##### Beta->GA

- [x] In use by multiple distributions
- [x] Approved by PRR and scalability

> - [x] Known bugs fixed

Reviewer: ```suggestion
- [x] Any known bugs fixed
```

Author: done

- [x] Tests passing
  - [x] E2E test [ServiceAccounts should mount projected service account token when requested](https://k8s-testgrid.appspot.com/sig-auth-gce#gce)
  - [x] E2E test [ServiceAccounts should set ownership and permission when RunAsUser or FsGroup is present](https://k8s-testgrid.appspot.com/sig-auth-gce#gce)
  - [x] E2E test [ServiceAccounts should support InClusterConfig with token rotation](https://k8s-testgrid.appspot.com/sig-auth-gce#gce-slow)

> #### RootCAConfigMap

Reviewer: update "Will enabling / using this feature result in increasing size or count of the existing API objects?" PRR section, controller creates one additional configmap per namespace

Author: done

| Alpha | Beta | GA |
| --- | --- | --- |
| 1.13 | 1.20 | 1.21 |

#### Beta->GA

- [ ] In use by multiple distributions
- [ ] Approved by PRR and scalability

> - [ ] Known bugs fixed

Reviewer: ```suggestion
- [ ] Any known bugs fixed
```

Author: done

#### BoundServiceAccountTokenVolume

| Alpha | Beta | GA |
| --- | --- | --- |
| 1.13 | 1.21 | 1.22 |

##### Alpha->Beta

- [x] Known bugs fixed
  - [x] PodSecurityPolicies that allow secrets but not projected volumes will
    prevent the use of token volumes.
    - Fixed in https://github.com/kubernetes/kubernetes/pull/92006
  - [x] In-cluster clients that don’t reload service account tokens will start
    failing an hour after deployment.
    - Mitigation added in https://github.com/kubernetes/kubernetes/issues/68164
  - [x] Pods running as non root may not access the service account token.
    - Fixed in https://github.com/kubernetes/kubernetes/pull/89193

- [x] Tests passing
  - [x] Upgrade test [sig-auth-serviceaccount-admission-controller-migration](https://k8s-testgrid.appspot.com/sig-auth-gce#upgrade-tests)

> - [ ] RootCAConfigMap GA

Reviewer: also add `TokenRequest/TokenRequestProjection GA`?

Author: done

##### Beta -> GA Graduation

> - [ ] Allow multi issuers to enable non distruptive issuer change.

Reviewer: ```suggestion
- [ ] Allow kube-apiserver to recognize multiple issuers to enable non disruptive issuer change.
```

Author: done

> - [ ] New `ServiceAccount` admission controller WAI in Beta for >= 1 minor without

Reviewer: ```suggestion
- [ ] New `ServiceAccount` admission controller working as intended in Beta for >= 1 minor release without
```

Author: done

significant issues.

> ## Production Readiness Review Questionnaire

Reviewer: change "pods' service account tokens will not be long-lived" to "pods' service account tokens will expire after 1 year by default"

Reviewer: drop the "pods created while the feature was enabled will reference a configmap that can grow stale with the feature disabled." bit from the PRR, since we're separating and prereqing the configmap controller feature gate so a single-version rollback will still keep the configmap current

Author: done
